bin/plugin/commands/release.js

```diff
@@ -390,9 +387,11 @@ async function runPushGitChangesStep(
 				true,
 				abortMessage
 			);
+			// We need to force-push because we are effectivelly rebasing from master
```

Comment:
Since the release is cherry-picked into master, the SHA changes, thus, pushing to the release branch is not possible anymore without force-pushing. 

This seems to work well, but I don't know if there might be unforeseen side-effects to force-pushing to the release branch. I'd love some feedback here.

Reply:
I don't like this. we shouldn't be force pushing to master IMO, we could potentially lose merges that happened in the meantime.

Reply:
At the very least, this should be [`--force-with-lease`](https://git-scm.com/docs/git-push).

(Btw, everyone should use that option in their daily Git interactions — I recommend adding aliases to Git like ` fpush = push --force-with-lease` or `fp` or `pf` or whatever.)

Reply:
Our ideal scenario here would be to have all RC commits (alongside potential other, cherry-picked, ones) on the release branch, no? So we need to ensure that we start off from the point where we "left off" with the previous RC. I'm wondering what's currently happening that gets in the way of that?

Could it be that the `git.checkoutLocalBranch` call in `runReleaseBranchCreationStep` should only be called if the branch didn't exist before, and that we need to run `checkoutRemoteBranch` otherwise?

Reply:
> Since the release commit is cherry-picked into master, the SHA changes, thus, pushing to the release branch is not possible anymore without force-pushing in the case of subsequent RCs.

Are you sure that's the reason? :thinking: Wouldn't we run into the same problem when releasing a stable version after an RC?

Reply:
I'm starting to think it would make sense to change the criterion for the ternary that runs either `runReleaseBranchCreationStep` or `runReleaseBranchCheckoutStep` from `isRC` to `isPackageVersionRC`, since it captures better what we're trying to do here:

- If the `package.json` version is an RC, then we can assume that a release branch has been created.
- If it doesn't, we're at the start of a new release cycle and need to create that branch.

This will mean moving some of the version computing logic around, but I think it'll fall in place quite naturally.

Reply:
@ockham that makes sense since second/third RCs will just just use the existing branch with cherrypicked commits

Reply:
> I don't like this. we shouldn't be force pushing to master IMO, we could potentially lose merges that happened in the meantime.

Note that the force push here is not to master, but to the `releaseBranch`.